improver/utilities/extract_from_table.py

```diff
@@ -0,0 +1,182 @@
+        sorted_index = np.clip(sorted_index, 0, None)
+
+        return sorted_index
+
+    def extract_table_values(
+        self, table: DataFrame, columns_cube: iris.cube.Cube, row_cube: iris.cube.Cube
+    ) -> np.array:
+        """Extract values from the table based on the provided row and column cubes.
+            Args:
+                table:
+                    DataFrame representing the table from which values are extracted.
+                columns_cube:
+                    Cube used to index the columns of the table.
+                row_cube:
+                    Cube used to index the rows of the table.
+            Returns:
+                Array of values extracted from the table.
+        """
+        shape = columns_cube.shape
+
+        columns_data = columns_cube.data.flatten()
+        row_data = row_cube.data.flatten()
+        column_index = self.nearest_lower_index(
+            values=columns_data, sorted_table_labels=table.columns
+        )
+        row_index = self.nearest_lower_index(
+            values=row_data, sorted_table_labels=table.index
+        )
+
+        result = table.to_numpy()[row_index, column_index]
+        result = result.reshape(shape)
+        return result
+
+    def convert_dict_to_dataframe(self, table: dict) -> DataFrame:
+        """Converts a dictionary to a pandas DataFrame"""
+
+        table = DataFrame.from_dict(table)
+        table.columns = table.columns.astype(float)
+        table.index = table.index.astype(float)
+
+        table = table.reindex(sorted(table.columns), axis=1)
+        table = table.reindex(sorted(table.index), axis=0)
+
+        return table
+
+    def process(self, *cubes: List[Cube], table: dict):
+        """
+        Process the input cubes and extract values from a table based on the provided row and
+        column indices. The row name is used to identify the cube used for indexing the rows
+        of the table. The new name is an optional argument that can be used to rename the
+        resulting cube.
+
+        Args:
+            cubes:
+                Input cubes for indexing columns and rows of the table. Exactly 2 cubes should
+                be provided one which contains the values to extract from the rows and one for
+                the columns.
+            table:
+                A dictionary representing the table from which values are extracted. Dictionary
+                should be in the form:
+                {"data":{column_name_1:{row_name_1:value, row_name_2:value},...},
+                "metadata":{"units":table_units}}
+                Other metadata can be included in the metadata dictionary such as a title for
+                the table but this will be ignored.
+
+        Returns:
+            Cube of the same shape and metadata as the row input cubes with values extracted
+            from the table based on the row and column input cubes. The cube will be re-named
+            if new_name is provided.
```

Comment:
Might say something about the unit setting here as well.

Reply:
Done